File: idealingua/idealingua-runtime-rpc-typescript/src/main/resources/runtime/typescript/irt/transport/auth/auth.custom.ts
import { AuthMethod } from './auth.method';

export class AuthCustom extends AuthMethod {
    public value: string;
    constructor(value: string) {
        super();
        this.value = value;
    }

    public fromValue(value: string): Error {
        this.value = value;
        return undefined;
    }

    public toValue(): string {
        return this.value;
    }
}

File: idealingua-v1/idealingua-v1-runtime-rpc-csharp/src/main/resources/runtime/csharp/IRT/Transport/Authorization/AuthCustom.cs

namespace IRT.Transport.Authorization {
    class AuthCustom: AuthMethod {
        public string Value;

        public AuthCustom(string value = null) {
            Value = value;
        }

        public override bool FromValue(string value) {
            Value = value;
            return true;
        }

        public override string ToValue() {
            return Value;
        }
    }
}